jgrapht-core/src/main/java/org/jgrapht/alg/isomorphism/ColorRefinementIsomorphismInspector.java

```diff
@@ -0,0 +1,299 @@
+     * returns the mapping if the calculated coarse coloring returned by color refinement is discrete.
+     * Otherwise it returns null.
+     *
+     * @return the mapping of the graph isomorphism if the coloring is discrete, otherwise null.
+     */
+    @Override
+    public Iterator<GraphMapping<V, E>> getMappings() throws IllegalStateException {
+        if(!isomorphismTestExecuted) {
+            try {
+                isomorphismExistsWithUndecidableCase();
+            } catch(IsomorphismUndecidableException ignored) {
+
+            }
+        }
+        if(isIsomorphic != null && isIsomorphic) {
+            ArrayList<GraphMapping<V, E>> iteratorList = new ArrayList<>(1);
+            iteratorList.add(isomorphicGraphMapping);
+            return iteratorList.iterator();
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public boolean isomorphismExists() {
+        try {
+            throw new IllegalAccessException("This method is deprecated and is undefined in this IsomorphismInspector!");
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    /**
+     * Check if an isomorphism exists.
+     *
+     * @return Optional.of(true) if there is an isomorphism, Optional.of(false) if there is no isomorphism,
+     * Optional.empty if ut cannot be decided whether there is an isomorphism or not.
+     */
+    @Override
+    public boolean isomorphismExistsWithUndecidableCase() throws IsomorphismUndecidableException {
+        if(isomorphismTestExecuted) {
+            if(isIsomorphic != null) {
+                return isIsomorphic;
+            } else {
+                throw new IsomorphismUndecidableException();
+            }
+        }
+
+        if(graph1.vertexSet().size() != graph2.vertexSet().size()) {
+            return false;
+        }
+
+        ColorRefinementAlgorithm<V, E> colorRefinementAlgorithm1 = new ColorRefinementAlgorithm<>(graph1);
+        ColorRefinementAlgorithm<V, E> colorRefinementAlgorithm2 = new ColorRefinementAlgorithm<>(graph2);
+
+        Coloring<V> coloring1 = colorRefinementAlgorithm1.getColoring(); // execute color refinement for graph1
+        Coloring<V> coloring2 = colorRefinementAlgorithm2.getColoring(); // execute color refinement for graph2
+
+        isIsomorphic = coarseColoringAreEqual(coloring1, coloring2);
+        return isIsomorphic;
+    }
+
+    /**
+     * returns whether the coarse colorings of the two given graphs are discrete if the colorings are the same.
+     * Otherwise, we do not computed if the coloring is discrete, hence, this method is undefined.
+     *
+     * @return if the both colorings are discrete if they are the same on both graphs.
+     */
+    public boolean isColoringDiscrete() {
+        if(!isomorphismTestExecuted) {
+            try {
+                isomorphismExistsWithUndecidableCase();
+            } catch(IsomorphismUndecidableException ignored) {
```

Comment:
Make the exception extend `RuntimeException` and throw it. Add the behaviour in the javadocs.